### PR TITLE
ARROW-11612: [C++] Rebuild trimmed boost bundle for 1.75.0

### PR DESCRIPTION
**Pull request by nealrichardson**
The trimmed bundle has been uploaded to the GitHub release, and I have confirmed locally downloading and building with it. The resulting package is slightly larger than we had before, even though I removed boost::regex and some headers that Thrift depended on before, because I added boost/process.hpp, which some tests appear to use (cf. ARROW-8771).

### CODE REVIEW

File: cpp/build-support/trim-boost.sh [hunk 1]
```diff
@@ -22,7 +22,15 @@
 # so that we don't have to download the whole big boost project when we build
 # boost from source.
 #
-# After running this script, run upload-boost.sh to put the bundle on bintray
+# To test building Arrow locally with the boost bundle this creates, add:
+#
+#     set(BOOST_SOURCE_URL /path/to/arrow/cpp/build-support/boost_1_75_0/boost_1_75_0.tar.gz)
+#
+# to the beginning of the build_boost() macro in ThirdpartyToolchain.cmake
```
Comment (kou) [inline cpp/build-support/trim-boost.sh:29]: `ARROW_BOOST_URL=/path/to/arrow/cpp/build-support/boost_1_75_0/boost_1_75_0.tar.gz cmake ...` will work.

**Review by kou — APPROVED** [review]
+1

**Comment by kou** [comment]
@github-actions autotune

**Comment by nealrichardson** [comment]
There's something missing for Thrift on Windows still, I'm investigating

**Comment by kou** [comment]
Sorry, I forgot to check CI results.

**Comment by nealrichardson** [comment]
@kou I believe I've fixed the Windows issue now

**Comment by nealrichardson** [comment]
Integration test failure is the usual flaky thing, so this is passing. I'll merge unless you object @kou

**Comment by kou** [comment]
No objection.

**Pull request closed without merging**